protovalidate/internal/extra_func.py

```diff
@@ -161,28 +170,50 @@ def is_email(string: celtypes.Value) -> celpy.Result:
 
 
 def is_uri(string: celtypes.Value) -> celpy.Result:
-    url = urlparse.urlparse(str(string))
-    # urlparse correctly reads the scheme from URNs but parses everything
-    # after (except the query string) as the path.
-    if url.scheme == "urn":
-        if not (url.path):
-            return celtypes.BoolType(False)
-    elif not all([url.scheme, url.netloc, url.path]):
-        return celtypes.BoolType(False)
+    """Validate whether string is a valid URI.
```

Comment:
Is there a reason not to use the same doc as the source implementation?

```
/**
 * Returns true if the string is a URI, for example "https://example.com/foo/bar?baz=quux#frag".
 *
 * URI is defined in the internet standard RFC 3986.
 * Zone Identifiers in IPv6 address literals are supported (RFC 6874).
 */
```

Reply:
For this I was following that PEP proposal:

> The docstring is a phrase ending in a period. It prescribes the function or method’s effect as a command (“Do this”, “Return that”), not as a description; e.g. don’t write “Returns the pathname …”.

I suppose we could change this to say 'Return true if the string...', but since we're documenting return types in the `Returns:` section, that doesn't make a lot of sense. I don't think the comments across implementations should be verbatim since each language is going to have its own guidelines and styles. As long as it expresses the exact same information. For example, the full description below this line includes all the RFC info.

Reply:
Fair to follow the PEP on phrasing 👍